### ISSUE TITLE
HTML generation with line numbers 

### ISSUE THREAD

**Issue by alok8bb** [trishume/syntect#441]
Is it possible to add line numbers to the generated HTML? I [tried](https://stackoverflow.com/questions/40842277/create-line-numbers-on-pre-with-css-only) adding counter to `pre` tag but since a single line can have multiple spans it does not work correctly.

**Comment by DSchroer** [comment]
Agreed. Having a single span per line under `<pre>` would be enough here.

**Comment by BBaoVanC** [comment]
> I [tried](https://stackoverflow.com/questions/40842277/create-line-numbers-on-pre-with-css-only) adding counter to `pre` tag

I tried the same; this is what it looks like

![image](https://user-images.githubusercontent.com/17971474/198172540-35a7a705-6c8c-4b8a-ad1f-0eb9a44f8633.png)

Does anyone have ideas for how I could try and make my own function for this? Line numbering is pretty important for my app.

**Comment by alok8bb** [comment]
> > I [tried](https://stackoverflow.com/questions/40842277/create-line-numbers-on-pre-with-css-only) adding counter to `pre` tag
> 
> I tried the same; this is what it looks like
> 
> ![image](https://user-images.githubusercontent.com/17971474/198172540-35a7a705-6c8c-4b8a-ad1f-0eb9a44f8633.png)
> 
> Does anyone have ideas for how I could try and make my own function for this? Line numbering is pretty important for my app.
> 

Yes, that's the same problem I'm having. There's no solution for this as of now.

**Comment by BBaoVanC** [comment]
I'm actually using ClassedHTMLGenerator right now where it needs a for loop that loops through every line. I wonder if somehow I can add a span in the loop?

**Comment by trishume** [comment]
Note that HTML generators use only public APIs and are fairly short, so you can copy-paste the functions into your own app and then modify them to produce whatever HTML you desire.

Although I'd probably accept a PR that did a good job of adding some classed span around every line as an option.
